auth/tests/integration/smoke_test.py

```diff
@@ -43,3 +45,16 @@
     assert token.access_token is not None
     assert token.access_token_duration != 0
     assert token.access_token_acquired_at != datetime.datetime(1970, 1, 1)
+
+
+@pytest.mark.asyncio  # type: ignore
+async def test_sign_blob(creds: str) -> None:
+    data = 'Testing Can be confidential!'
+    payload = encode(data)
+
+    async with aiohttp.ClientSession(conn_timeout=10, read_timeout=10) as s:
+
+        iam_credentials = IamCredentialsClient(service_file=creds, session=s)
+        signed_data = iam_credentials.sign_blob(payload)
+        assert signed_data  # TODO(nick): How do we verify data is signed?
```

Comment:
Any ideas on this? I don't see any reversible API call we could check with.

Reply:
My best thought is to have the string presigned and checked in as test data. We could pre-sign it using the gcloud command line tool and check in the value and then confirm its the same?
So dont see anything about how to verify the data is signed by your service account in the docs.

Reply:
That seems reasonable? I ran your tests on CI, here's the signed blob:
```
Ye7i1BxgqYnHBC9mZxXU+uV0/r0zH4mvm7qWW166mmixSC00vC4qfrEnwipYG4QYA7A52KtBQcOWe9SnCYyw/shqlz9QJH8TW5I+f7/WaE+09GUBinp7bvHpolQ/G9hyYS0a4mUN4Yzh9UqsMfKbi/9chWKVrX8XlWrDxORv2a1F7M5rvr971AJ5jZDlmAVK5FHk48O3xBRGgIle36v+HgS7NLfoAMfyVrjYl1LFByd/uR+1GmGwTBi0LttpGfWm70TXxwUcijZ8BZcxkJGqtIslqceEukP1PVztAR8g8LTO6yfqoI5/tKlsffW5b7OdpJGQEYFv69OVHflvCFKPgQ==
```

Reply:
Well, either I did not reproduce the results correctly in the below command (@nickfirespotter could you please verify?) or the results change based on time. I'll re-run the below tomorrow to see if the later is the case.

```
$ gcloud iam service-accounts sign-blob \
    --iam-account=public-testing@voiceai-staging.iam.gserviceaccount.com \
    <(python3 -c "import base64; print(base64.b64encode('Testing Can be confidential!'.encode('utf-8')).replace(b'+', b'-').replace(b'/', b'_').decode('utf-8'), end='')") \
    ./foo  2>/dev/null
$ python3 -c "import base64; print(base64.b64encode(open('./foo', 'rb').read()).replace(b'+', b'-').replace(b'/', b'_').decode('utf-8'))"
FnmTO-oK-joLmPUgFgvSaSeBmabWen38W70-LlTSfCi1BfcKh24jTtLYnvTLR6qCcet7qyiFtwXoGHYbFgesWsA327NiO-X3EvXbakf9rW_c5wheQMFV4Hf0lmRJOYF7o7jGCgBY47DHlEo531stIXn80U_9zmnHcmEZn9o9lUKb7WccIvDLUS-gPcGCBL7UVjtrMtuM6sXOPe1dhhdKtLQhfar4oC4paCVWx4ksFIiPkAXyV5_uvrKplS2PC2Y5JVKUdRjBAq9Q3Cf_BWOMmpo5L93dGuPVM5bWmRjH6OPanLPr5JQg-j5MrRB-ENfMD8zwrhPwbN-eBRrFTjPbbQ==
```

Reply:
The returned blob is the same -- but it still doesn't match the signed blob the tests output.

Reply:
Removing the bytes fuckery, I get:
```
$ gcloud iam service-accounts sign-blob \
    --iam-account=public-testing@voiceai-staging.iam.gserviceaccount.com \
    <(python3 -c "import base64; print(base64.b64encode(b'Testing Can be confidential!').replace(b'+', b'-').replace(b'/', b'_'), end='')") \
    ./foo  2>/dev/null
$ python3 -c "import base64; print(base64.b64encode(open('./foo', 'rb').read()).replace(b'+', b'-').replace(b'/', b'_'))"
b'JaZ9SMwrNJN2dE9WsTn8ds6isQwzCRngmWUrEu8-m8fTt0-zOyArUE1cc1pS_tyMgcx_MKd3eBbLHbKZ4gF5RAMWczujX--YlO09e4Ot_UM_1WDO3bwA0iKlg_14jYEay6i2gJ7zOS4XfoCG4aotVyshzKoLRpMB98wbFp1AZxZo35q7UZH98Z4mtU5BLV40RVk7lGhZdvgXEgvbECdEMXSN1fI4kayKS7ks_KLJ4liPugJpMQDwvf7YJkx8xuvtZBc0exo63bFin8NDA-PMk6XNF_7OoEiuubr48I-7UItqOjXrqmAPnngXTDAt9AiYDG88_4bjP2yFLta1DIdhbA=='
```